lgc/builder/DescBuilder.cpp

```diff
@@ -424,10 +424,9 @@ Value *BuilderImpl::getDescPtr(ResourceNodeType concreteType, ResourceNodeType a
     // - descriptor binding number
     // - value for high 32 bits of the pointer; HighAddrPc to use PC
     if (node || topNode || concreteType != ResourceNodeType::DescriptorFmask) {
-      unsigned shadowDescriptorTable = m_pipelineState->getOptions().shadowDescriptorTable;
-      bool shadow =
-          concreteType == ResourceNodeType::DescriptorFmask && shadowDescriptorTable != ShadowDescriptorTableDisable;
-      Value *highHalf = getInt32(shadow ? shadowDescriptorTable : HighAddrPc);
+      unsigned highAddrOfFmask = m_pipelineState->getOptions().highAddrOfFmask;
+      bool isFmask = concreteType == ResourceNodeType::DescriptorFmask;
+      Value *highHalf = getInt32(isFmask ? highAddrOfFmask : HighAddrPc);
```

Comment:
Please add an
```
assert(!(isFmask && highAddrOfFmask == ShadowDescriptorTableDisable) && "not implemented");
```
After all, in the longer term this change only makes sense if it is possible to use fmasks without a shadow descriptor table.